Sources/Target/Common/ProcessBase.cpp

```diff
@@ -288,6 +284,26 @@ ErrorCode ProcessBase::afterResume() {
   return kSuccess;
 }
 
+SoftwareBreakpointManager *ProcessBase::softwareBreakpointManager() const {
+#if defined(ARCH_ARM) || defined(ARCH_ARM64)
+  typedef Architecture::ARM::SoftwareBreakpointManager SWBPMType;
+#elif defined(ARCH_X86) || defined(ARCH_X86_64)
+  typedef Architecture::X86::SoftwareBreakpointManager SWBPMType;
+#endif
+  static SWBPMType softwareBreakpointManager(const_cast<ProcessBase *>(this));
```

Comment:
Good point. I'll go back to the previous method.

On Thursday, April 28, 2016, strager <notifications@github.com> wrote:

> In Sources/Target/Common/ProcessBase.cpp
> <https://github.com/facebook/ds2/pull/304#discussion_r61447703>:
>
> > @@ -288,6 +284,26 @@ ErrorCode ProcessBase::afterResume() {
> >    return kSuccess;
> >  }
> >
> > +SoftwareBreakpointManager *ProcessBase::softwareBreakpointManager() const {
> > +#if defined(ARCH_ARM) || defined(ARCH_ARM64)
> > +  typedef Architecture::ARM::SoftwareBreakpointManager SWBPMType;
> > +#elif defined(ARCH_X86) || defined(ARCH_X86_64)
> > +  typedef Architecture::X86::SoftwareBreakpointManager SWBPMType;
> > +#endif
> > +  static SWBPMType softwareBreakpointManager(const_cast<ProcessBase *>(this));
>
> I don't like this. This implies I cannot debug multiple processes at once
> (with one ds2 instance). Why can't this be a member variable?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/ds2/pull/304/files/1d994978cfc86b0aad896d0c34fd2d79cacefe9a..e7e1963afb287907b44f6accbe4a37ce9db0b078#r61447703>
>


-- 
Stephane Sezer